firestore.rules

```diff
@@ -8,20 +8,31 @@ function didSign() {
   return request.auth.token.email == resource.data.signer.value;
 }
 
+// the request object contains info about the authentication status of the requesting user
+// if the .auth property is not set, the user is not signed in
+function isSignedIn() {
+  return request.auth != null;
+}
+
+// checks whether the user has admin privileges or not
+function isAdmin() {
+    return request.auth.token.admin == true;
+}
+
 // Note: CLAs and Addendums cannot be modified or deleted
 service cloud.firestore {
   match /databases/{database}/documents {
     match /agreements/{agreementId} {
-      allow create: if request.auth != null && isSigner();
-      allow read:   if request.auth != null && didSign(); //TODO (see below #2)
+      allow create: if isSignedIn() && isSigner();
+      allow read:   if isSignedIn() && (didSign() || isAdmin()); //TODO (see below #2)
```

Comment:
Just a token of appreciation for firestore and whoever designed the permission rule scheme. It's great.